src/compass/utils/validate_cslc.py

```diff
@@ -0,0 +1,136 @@
+    file1: str
+        File path to reference CSLC product (golden dataset)
+    file2: str
+        File path to generated CSLC product to use for comparison
+    '''
+
+    # Check if file paths exits
+    if not os.path.exists(file_ref):
+        print(f'ERROR reference CSLC product not found: {file_ref}')
+        return
+
+    if not os.path.exists(file_sec):
+        print(f'ERROR secondary CSLC product not found: {file_sec}')
+        return
+
+    # Extract some info from reference/secondary CSLC products
+    dataset_ref = gdal.Open(file_ref, gdal.GA_ReadOnly)
+    geotransform_ref = dataset_ref.GetGeoTransform()
+    nbands_ref = dataset_ref.RasterCount
+
+    dataset_sec = gdal.Open(file_sec, gdal.GA_ReadOnly)
+    geotransform_sec = dataset_sec.GetGeoTransform()
+    nbands_sec = dataset_sec.RasterCount
+
+    # Compare number of bands
+    print('Comparing CSLC number of bands ...')
+    if not nbands_ref == nbands_sec:
+        print(f'ERROR Number of bands in reference CSLC {nbands_ref} differs'
+              f'from number of bands {nbands_sec} in secondary CSLC')
+        return
+
+    print('Comparing geo transform arrays ...')
+    if not np.array_equal(geotransform_ref, geotransform_sec):
+        print(f'ERROR Reference geo transform array {dataset_ref} differs'
+              f'from secondary CSLC geo transform array {dataset_sec}')
+        return
+
+    # Compare amplitude of reference and generated CSLC products
+    slc_ref = dataset_ref.GetRasterBand(1).ReadAsArray()
+    slc_sec = dataset_sec.GetRasterBand(1).ReadAsArray()
+
+    print('Check mean real part difference between CSLC products is < 1.0e-5')
+    assert np.allclose(slc_ref.real, slc_sec.real,
+                       atol=0.0, rtol=1.0e-5, equal_nan=True)
+    print('Check mean imaginary part difference between CSLC products is < 1.0e-5')
+    assert np.allclose(slc_ref.imag, slc_sec.imag,
+                       atol=0.0, rtol=1.0e-5, equal_nan=True)
+
+    return
+
+
+def compare_cslc_metadata(file_ref, file_sec):
+    '''
+    Compare reference and generated CSLC metadata
+    '''
+
+    # Check if metadata files exists
+    if not os.path.exists(file_ref):
+        print(f'ERROR reference CSLC metadata not found: {file_ref}')
+        return
+
+    if not os.path.exists(file_sec):
+        print(f'ERROR CSLC metadata not found: {file_sec}')
+        return
+
+    # Load metadata
+    with open(file_ref, 'r') as f:
+        metadata_ref = json.load(f)
+
+    with open(file_sec, 'r') as f:
+        metadata_sec = json.load(f)
+
+    # Intersect metadata keys
+    set_ref_minus_sec = metadata_ref.keys() - metadata_sec.keys()
+    set_sec_minus_ref = metadata_sec.keys() - metadata_ref.keys()
+
+    err_str = "Metadata keys do not match.\n"
+    if set_ref_minus_sec:
+        err_str += f'Reference CSLC metadata extra entries: {set_ref_minus_sec}'
+    if set_sec_minus_ref:
+        err_str += f'Secondary CSLC metadata extra entries: {set_sec_minus_ref}'
```

Comment:
```suggestion
    err_str = "Metadata keys do not match."
    if set_ref_minus_sec:
        err_str += f'\nReference CSLC metadata extra entries: {set_ref_minus_sec}'
    if set_sec_minus_ref:
        err_str += f'\nSecondary CSLC metadata extra entries: {set_sec_minus_ref}'
```
Put everything on different line for readability.